src/machine/machine_esp32.go

```diff
@@ -363,38 +363,16 @@ func (spi SPI) Configure(config SPIConfig) error {
 	// (mode), among others.
 	var ctrl2Reg, userReg, pinReg uint32
 	// For mode configuration, see table 29 in the reference manual (page 128).
-	var delayMode uint32
 	switch config.Mode {
 	case 0:
-		delayMode = 2
 	case 1:
-		delayMode = 1
 		userReg |= esp.SPI_USER_CK_OUT_EDGE
 	case 2:
-		delayMode = 1
 		userReg |= esp.SPI_USER_CK_OUT_EDGE
 		pinReg |= esp.SPI_PIN_CK_IDLE_EDGE
 	case 3:
-		delayMode = 2
 		pinReg |= esp.SPI_PIN_CK_IDLE_EDGE
 	}
-	// Extra configuration necessary for correct data input at high frequencies.
-	// This is only necessary when MISO goes through the GPIO matrix (which it
-	// currently does).
-	if config.Frequency >= 40e6 {
-		// Delay mode must be set to 0 and SPI_USR_DUMMY_CYCLELEN should be set
-		// to 0 (the default).
-		userReg |= esp.SPI_USER_USR_DUMMY
```

Comment:
From the datasheet:

> If GP-SPI output clock frequency is clkapb/2, register SPI_MISO_DELAY_MODE should be set to 0 and the dummy phase should be enabled (SPI_USR_DUMMY = 1) for one clkspi clock cycle (SPI_USR_DUMMY_CYCLELEN = 0) when configuring the clock polarity;

So the datasheet says the `SPI_USER_USR_DUMMY` bit must be set.

I wonder if the datasheet is wrong or something else is going on?

Reply:
```go
		// Delay mode must be set to 0 and SPI_USR_DUMMY_CYCLELEN should be set
		// to 0 (the default).
```

![image](https://user-images.githubusercontent.com/9251039/134474077-5fa5b92c-5643-46eb-9dd8-3e0d4c973e7a.png)


The source code says that the initial value of SPI_USR_DUMMY_CYCLELEN is 0.
However, in TRM, ResetValue=7.
By adding the following initialization, the dummy cycle is no longer output to the bus.

```go
spi.Bus.USER1.Set(0)
```

Reply:
Added initialization of SPI_USR_DUMMY_CYCLELEN.

https://github.com/tinygo-org/tinygo/blob/86a4c71ebbe07d4399520589d039951edf0eada2/src/machine/machine_esp32.go#L387